pkg/utils/build/source_image.go

```diff
@@ -0,0 +1,164 @@
+package build
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/redhat-appstudio/e2e-tests/pkg/utils"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+)
+
+const (
+	extraSourceSubDir     = "extra_src_dir"
+	rpmSubDir             = "rpm_dir"
+	srcTarFileRegex       = "extra-src-[0-9]+.tar"
+	shaValueRegex         = "[a-f0-9]{40}"
+	tarGzFileRegex        = ".tar.gz$"
+	gomodDependencySubDir = "deps/gomod/pkg/mod/cache/download/"
+	pipDependencySubDir   = "deps/pip/"
+)
+
+func GetBinaryImage(pr *v1beta1.PipelineRun) string {
+	for _, p := range pr.Spec.Params {
+		if p.Name == "output-image" {
+			return p.Value.StringVal
+		}
+	}
+	return ""
+}
+
+func IsSourceBuildEnabled(pr *v1beta1.PipelineRun) bool {
```

Comment:
Does it make sense to have `getPipelineParameter` method?

Reply:
Could you please explain a bit more here. How do you want to see the `getPipelineParameter` method used.

Reply:
I see that we have at least two methods that read a parameter value from the pipeline. If it's only two, we can leave it as is. But if it's more or we use it on different suites, then maybe we should consider an utility function that read parameter value bu parameter name?

Reply:
Okay got it .. will prefer to keep it as it is for now.. in case we have more such method reading pipeline parameter value will address it then